db: TestCleaner fails under the race build on master on linux

```
arjunnair@gceworker-arjunnair:~/go/src/github.com/cockroachdb/pebble$ git branch
* master
  ve_enc_dec
arjunnair@gceworker-arjunnair:~/go/src/github.com/cockroachdb/pebble$ go test -race -exec 'stress' -run ^TestCleaner$

--- FAIL: TestCleaner (0.02s)
    datadriven.go:143: 
        testdata/cleaner:196:
         
        output didn't match expected:
        @@ -33,6 +33,10 @@
         remove: db1_wal/000002.log
         remove: db1_wal/000004.log
         remove: db1/000123.sst
         remove: db1/000456.sst
         remove: db1/OPTIONS-000003
        +open: db1/000005.sst
        +read-at(744, 53): db1/000005.sst
        +read-at(707, 37): db1/000005.sst
        +read-at(79, 628): db1/000005.sst
         
FAIL


ERROR: exit status 1

1 runs completed, 1 failures, over 0s
context canceled
FAIL
exit status 1
FAIL    github.com/cockroachdb/pebble   0.074s
```

This started failing more regularly due to https://github.com/cockroachdb/pebble/pull/2484, but either way, it exists on master.

--- a/cleaner_test.go
+++ b/cleaner_test.go
@@ -107,6 +107,8 @@ func TestCleaner(t *testing.T) {
 					return "open <dir> [archive] [readonly]"
 				}
 			}
+			// Asynchronous table stats retrieval makes the output flaky.
+			opts.private.disableTableStats = true
 			d, err := Open(dir, opts)
 			if err != nil {
 				return err.Error()

